site/app/libraries/database/DatabaseQueries.php

```diff
@@ -2263,7 +2302,7 @@ public function existsAnnouncements($show_deleted = false){
     }
 
     public function viewedThread($user, $thread_id){
-      $this->course_db->query("SELECT * from viewed_responses where thread_id = ? and user_id = ?", array($thread_id, $user));
+      $this->course_db->query("SELECT * FROM viewed_responses v WHERE thread_id = ? AND user_id = ? AND NOT EXISTS(SELECT thread_id FROM (posts LEFT JOIN forum_posts_history ON posts.id = forum_posts_history.post_id) AS jp WHERE jp.thread_id = ? AND (jp.timestamp > v.timestamp OR (jp.edit_timestamp IS NOT NULL AND jp.edit_timestamp > v.timestamp)))" , array($thread_id, $user, $thread_id));
```

Comment:
Do we need to also include edits? Not sure if this was discussed or not. If not, we can simplify this query to just be: `select 1 from viewed_responses v, posts p where v.thread_id = 6 and v.user_id='instructor' and p.timestamp > v.timestamp;`

Reply:
We should also include edits, please disregard this comment.